File: summer2019/week2/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        // Create a hashset of all jewels for O(1) access
        HashSet<Character> set = new HashSet<>();
        for (int i = 0; i < J.length(); i++) {
            set.add(J.charAt(i));
        }

        // Iterate through S and count number of
        // characters also in set
        int numJewels = 0;
        for (int i = 0; i < S.length(); i++) {
            if (set.contains(S.charAt(i))) {
                numJewels++;
            }
        }
        return numJewels;
    }
}

File: spring2020/unit2/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        jewels = set()
        for j in J:
            jewels.add(j)
        num = 0
        for s in S:
            if s in jewels:
                num += 1
        return num
